extras/schemas/resin-os.json

```diff
@@ -0,0 +1,185 @@
+                                }
+                            }
+                        ]
+                    },
+                    "pubnub": {
+                        "properties": [
+                            {
+                                "definition": {
+                                    "publishKey": {
+                                        "type": ["string"],
+                                        "mapping": ["/pubnubPublishKey"]
+                                    },
+                                    "subscribeKey": {
+                                        "type": ["string"],
+                                        "mapping": ["/pubnubSubscribeKey"]
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        ]
+    },
+    "resin_wifi": {
+        "type": "ini",
+        "location": { 
+            "partition": 1,
+            "path": ["system-connections", "resin-wifi-01"]
+        },
+        "properties": [{   
+            "definition": {
+                "wifi": {
+                    "properties": [{
+                        "definition": {
+                            "enable": {
+                                "type": ["boolean"],
+                                "mapping": [
+                                    {
+                                        "value": true,
+                                        "template": {
+                                            "connection": {
+                                                "id": "resin-wifi-01",
+                                                "type": "wifi"
+                                            },
+                                            "wifi": {
+                                                "mode": "infrastructure",
+                                                "ssid": "[[string]]"
+                                            },
+                                            "wifi-security": {
+                                                "auth-alg": "open",
+                                                "key-mgmt": "wpa-psk",
+                                                "psk": "[[string]]"
+                                            },
+                                            "ipv4": {
+                                                "method": "auto"
+                                            },
+                                            "ipv6": {
+                                                "addr-gen-mode": "stable-privacy",
+                                                "method": "auto"
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "value": false,
+                                        "template": {}
```

Comment:
It'd be nice to test this on an actual resin device. I want to make sure NetworkManager will not get confused by an empty file.